lib/src/components/NxTextInput/__tests__/NxTextInput.test.tsx

```diff
@@ -4,258 +4,518 @@
+        el = renderEl({ ref });
 
-    expect(invalidMessage).toHaveProp('role', 'alert');
-    expect(invalidMessageId).toBeTruthy();
-    expect(component.find('.nx-text-input__input')).toHaveProp('aria-errormessage', invalidMessageId);
+    expect(ref.current).toBe(el);
   });
 
-  it('sets aria-invalid to true if it is validatable and has validation errors, ' +
-      'and either isPristine is false or a surrounding FormAriaContext\'s showValidationErrors is true', function() {
-    expect(getShallowComponent().find('input')).not.toHaveProp('aria-invalid', true);
-    expect(getShallowComponent({ validatable: true }).find('input')).not.toHaveProp('aria-invalid', true);
-    expect(getShallowComponent({ validatable: true, validationErrors: 'foo' }).find('input'))
-        .toHaveProp('aria-invalid', true);
-
-    expect(getShallowComponent({ isPristine: true, validatable: true, validationErrors: 'foo' }).find('input'))
-        .not.toHaveProp('aria-invalid', true);
-
-    function getMountedWithContext(extraProps: Partial<Props> = {}) {
-      return mount(
-        <FormAriaContext.Provider value={{ showValidationErrors: true }}>
-          <PrivateNxTextInput { ...minimalProps } { ...extraProps } />
-        </FormAriaContext.Provider>
-      );
-    }
+  it('adds specified classNames to the top-level element in addition to the defaults', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
 
-    expect(getMountedWithContext().find('input')).not.toHaveProp('aria-invalid', true);
-    expect(getMountedWithContext({ validatable: true }).find('input')).not.toHaveProp('aria-invalid', true);
-    expect(getMountedWithContext({ validatable: true, validationErrors: 'foo' }).find('input'))
-        .toHaveProp('aria-invalid', true);
+    expect(el).toHaveClass('foo');
 
-    expect(getMountedWithContext({ isPristine: true, validatable: true, validationErrors: 'foo' }).find('input'))
-        .toHaveProp('aria-invalid', true);
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('renders a text input by default', function() {
-    expect(getShallowComponent().find('input')).toHaveProp('type', 'text');
-  });
+  it('passes additional attrs to the input', function() {
+    const input = quickRender({ id: 'foo', lang: 'en-US' }).getByRole('textbox');
 
-  it('renders a password input if type is "password"', function() {
-    expect(getShallowComponent({ type: 'password' }).find('input')).toHaveProp('type', 'password');
+    expect(input).toHaveAttribute('id', 'foo');
+    expect(input).toHaveAttribute('lang', 'en-US');
   });
 
-  it('renders a date input if type is "date"', function() {
-    expect(getShallowComponent({ type: 'date' }).find('input')).toHaveProp('type', 'date');
+  it('sets the value as specified', function() {
+    expect(quickRender({ value: 'boo' }).getByRole('textbox')).toHaveAttribute('value', 'boo');
```

Comment:
The attribute isn't what matters to us and is not kept in sync with the actual `value` DOM property
```suggestion
    expect(quickRender({ value: 'boo' }).getByRole('textbox')).toHaveValue('boo');
```